include/na/nasp/CodeGenerator.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "Solver.hpp"
+#include "ir/QuantumComputation.hpp"
+#include "na/NAComputation.hpp"
+#include "na/NADefinitions.hpp"
+
+#include <cstdint>
+
+namespace na {
+using namespace qc;
+
+class CodeGenerator {
+private:
+  static auto coordFromDiscrete(std::int32_t x, std::int32_t y, std::int32_t h,
+                                std::int32_t v, std::int32_t maxHOffset,
+                                std::int32_t maxVOffset,
+                                std::int32_t minEntanglingY,
+                                std::int32_t maxEntanglingY) -> Point;
+
+public:
+  [[nodiscard]] static auto
+  generate(const QuantumComputation& input, const NASolver::Result& result,
+           std::uint16_t maxHOffset, std::uint16_t maxVOffset,
+           std::uint16_t minEntanglingY, std::uint16_t maxEntanglingY)
```

Comment:
That is because I want to use the NAComputation in the end which uses the class Point to store the position of the atoms. This class uses int64 coordinates. However, the solver cannot deal with large coordinates anyways. Nevertheless, I found your comment regarding the casts reasonable and will use int64 now from the start.